flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1179,6 +1179,14 @@ protected Optional<WorkerType> closeTaskManagerConnection(
             slotManager.unregisterTaskManager(workerRegistration.getInstanceID(), cause);
             clusterPartitionTracker.processTaskExecutorShutdown(resourceID);
 
+            jobManagerRegistrations
+                    .values()
+                    .forEach(
+                            jobManagerRegistration ->
+                                    jobManagerRegistration
+                                            .getJobManagerGateway()
+                                            .disconnectTaskManager(resourceID, cause));
```

Comment:
It's better to add a test to check whether the `disconnectTaskManager` is called for JobMaster.

Reply:
@1996fanrui I have added a unit test.